src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultRazorDynamicFileInfoProvider.cs

```diff
@@ -343,7 +343,10 @@ private ProjectId TryFindProjectIdForProjectKey(ProjectKey key)
 
     private RazorDynamicFileInfo CreateEmptyInfo(Key key)
     {
-        var projectKey = TryFindProjectKeyForProjectId(key.ProjectId).AssumeNotNull();
+        // We have to allow null here, because we are activated for all Roslyn projects that have a .cshtml or .razor file
+        // and our Roslyn EA doesn't allow null returns, so we need to return something. In these situations we end up
```

Comment:
Can we file a bug to fix this in the future?

Reply:
Yeah, I'm going to open a Roslyn PR for this side of things today, as there is no need for us to be called at all for non-C# projects.

Reply:
Opened https://github.com/dotnet/roslyn/pull/69828